commit 0df9058960d3fa24aa7695a4a14524127cc0e9be
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Tue Apr 21 08:44:52 2015 +0200

    Fix tsInfo.conditionals in deselect() when the package is not yet in the transaction BZ#1168385

diff --git a/AUTHORS b/AUTHORS
index 71845e1..350e136 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -24,6 +24,7 @@ YUM AUTHORS
     James Antill
     Panu Matilainen
     Tambet Ingo
+    Valentina Mukhamedzhanova
 
 
 Original Yup people:
diff --git a/yum/transactioninfo.py b/yum/transactioninfo.py
index 9ce5025..ec2c7cb 100644
--- a/yum/transactioninfo.py
+++ b/yum/transactioninfo.py
@@ -237,6 +237,7 @@ class TransactionData:
         if not txmbrs:
             if self._inSack is not None:
                 pkgs = self._inSack.returnPackages(patterns=[pattern])
+            if pkgs: pass
             elif self.pkgSack is None:
                 pkgs = []
             else:
commit c8b564968286ffccbd46beeb40c95d5e1c74a2aa
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Mon Jul 13 13:45:45 2015 +0200

    Fix UnboundLocalError in deselect()

diff --git a/yum/transactioninfo.py b/yum/transactioninfo.py
index ec2c7cb..8d47caa 100644
--- a/yum/transactioninfo.py
+++ b/yum/transactioninfo.py
@@ -235,13 +235,13 @@ class TransactionData:
                 txmbrs = self.matchNaevr(na[0], na[1])
 
         if not txmbrs:
+            pkgs = []
             if self._inSack is not None:
                 pkgs = self._inSack.returnPackages(patterns=[pattern])
-            if pkgs: pass
-            elif self.pkgSack is None:
-                pkgs = []
-            else:
+
+            if not pkgs and self.pkgSack is not None:
                 pkgs = self.pkgSack.returnPackages(patterns=[pattern])
+
             if not pkgs:
                 pkgs = self.rpmdb.returnPackages(patterns=[pattern])
 
